src/cls/rbd/cls_rbd.cc

```diff
@@ -7459,11 +7515,37 @@ int group_snap_set(cls_method_context_t hctx,
     } else if (r >= 0) {
       return -EEXIST;
     }
+
+    uint64_t max_order = 0;
+    std::string max_order_key = group::RBD_GROUP_SNAP_MAX_ORDER_KEY;
+
+    r = read_key(hctx, max_order_key, &max_order);
+    if (r < 0 && r != -ENOENT) {
+      return r;
+    }
+    std::string order_key = group::snap_order_key(group_snap.id);
+    bufferlist bl;
+    encode(++max_order, bl);
+    r = cls_cxx_map_set_val(hctx, max_order_key, &bl);
+    if (r < 0) {
+      CLS_ERR("error setting key: %s : %s", max_order_key.c_str(),
+              cpp_strerror(r).c_str());
```

Comment:
Other functions log the error str as well. For example : `set_op_features`, `set_migration` etc. It doesn't do any harm and makes debugging easier in case the error code has not been captured elsewhere.

Reply:
You have a point.  There are also functions that don't log it, but in `src/cls/rbd/cls_rbd.cc` it seems to be more common to log than not.  I'll remove my other comment that concerned just calling `cpp_strerror()`.